utils/db.ts

```diff
@@ -171,6 +171,109 @@ export async function getItemsSince(msAgo: number) {
   });
 }
 
+// Notification
+export interface Notification {
+  userId: string;
+  type: string;
+  userFromId: string;
+  userFromLogin: string;
+  originId: string;
+  originTitle: string;
+  // The below properties can be automatically generated upon item creation
+  id: string;
+  createdAt: Date;
+}
+
+export function newNotificationProps(): Pick<Item, "id" | "createdAt"> {
+  return {
+    id: crypto.randomUUID(),
+    createdAt: new Date(),
+  };
+}
+
+/**
+ * Creates a new notification in KV. Throws if the item already exists in one of the indexes.
+ *
+ * @example New notification creation
+ * ```ts
+ * import { newNotificationProps, createNotification } from "@/utils/db.ts";
+ *
+ * const notification: Notification = {
+ *   userId: "example-user-id",
+ *   type: "example-type",
+ *   userFromId: "example-user-from-id"
+ *   userFromLogin: "example-user-from-login"
+ *   originId: "example-origin-id"
+ *   originTitle: "example-origin-title"
+ *   ...newNotificationProps(),
+ * };
+ *
+ * await createNotification(notification);
+ * ```
+ */
+export async function createNotification(notif: Notification) {
+  const notificationsKey = ["notifications", notif.id];
+  const notificationsByTimeKey = [
+    "notifications_by_time",
+    notif.createdAt.getTime(),
+    notif.id,
+  ];
+  const notificationsByUserKey = [
+    "notifications_by_user",
+    notif.userId,
+    notif.id,
+  ];
+
+  const res = await kv.atomic()
+    .check({ key: notificationsKey, versionstamp: null })
+    .check({ key: notificationsByTimeKey, versionstamp: null })
+    .check({ key: notificationsByUserKey, versionstamp: null })
+    .set(notificationsKey, notif)
+    .set(notificationsByTimeKey, notif)
+    .set(notificationsByUserKey, notif)
+    .commit();
+
+  if (!res.ok) throw new Error(`Failed to create notification: ${notif}`);
+}
+
+export async function deleteNotification(notif: Notification) {
+  const notificationsKey = ["notifications", notif.id];
+  const notificationsByTimeKey = [
+    "notifications_by_time",
+    notif.createdAt.getTime(),
+    notif.id,
+  ];
+  const notificationsByUserKey = [
+    "notifications_by_user",
+    notif.userId,
+    notif.id,
+  ];
+
+  const res = await kv.atomic()
+    .delete(notificationsKey)
+    .delete(notificationsByTimeKey)
+    .delete(notificationsByUserKey)
+    .commit();
+
+  if (!res.ok) throw new Error(`Failed to delete notification: ${notif}`);
+}
+
+export async function getNotification(id: string) {
+  return await getValue<Notification>(["notifications", id]);
+}
+
+export async function getNotificationsByUser(userId: string) {
+  return await getValues<Notification>({
+    prefix: ["notifications_by_user", userId],
+  });
+}
+
+export async function getNotificationsCountByUser(userId: string) {
+  const notificationsCountByUser =
+    (await getNotificationsByUser(userId)).length;
+  return notificationsCountByUser;
+}
```

Comment:
ok! I've done the changes. But just so I learn: why choosing to do this here?

Reply:
If we use a dot instead of a count, the exact number of unread notifications matters less. What matters is _if_ there are any unread notifications at all. The eventual consistency mode may return a stale value, but it is faster.